yt_dlp/extractor/ceskatelevize.py

```diff
@@ -179,25 +227,25 @@ def _real_extract(self, url):
             if not playlistpage:
                 continue
 
-            playlist_url = playlistpage['url']
-            if playlist_url == 'error_region':
-                raise ExtractorError(NOT_AVAILABLE_STRING, expected=True)
-
-            req = Request(urllib.parse.unquote(playlist_url))
-            req.headers['Referer'] = url
-
-            playlist = self._download_json(req, playlist_id, fatal=False)
-            if not playlist:
-                continue
+            playlist_url = playlistpage.get('url')
+            if playlist_url:
+                if playlist_url == 'error_region':
+                    raise ExtractorError(NOT_AVAILABLE_STRING, expected=True)
+                req = Request(urllib.parse.unquote(playlist_url))
+                req.headers['Referer'] = url
+                playlist = self._download_json(req, playlist_id, fatal=False)
+                if not playlist:
+                    continue
+                playlist = playlist.get('playlist')
+            else:
+                playlist = traverse_obj(playlistpage, ('RESULT', 'playlist'))
 
-            playlist = playlist.get('playlist')
             if not isinstance(playlist, list):
                 continue
 
             playlist_len = len(playlist)
```

Comment:
This variable is used in a rather obscure test at new l.268 that really belongs outside the loop over `playlist`. It's saying that when `user_agent` the `entries` may have been populated; if so, and the `playlist` fetched with that UA has the same length, the existing formats (may be `[]`) are extended, whereas if the `playlist` doesn't have the same length, a new entry is appended with the formats from this iteration.

Instead, set this (which will be `False` on the first iteration of `user_agent`):
```suggestion
            extending = len(entries) == len(playlist)
```
and at l.268:
```diff
-                if user_agent and len(entries) == playlist_len:
+                if extending:
                     entries[num]['formats'].extend(formats)
                     continue
```